src/main/java/com/ezasm/simulation/Register.java

```diff
@@ -70,8 +72,10 @@ public void setData(RawData data) {
     }
 
     public void setData(byte[] data) {
```

Comment:
we don't pass data around as `byte[]`s anymore. There's a method call above this which uses a new class `RawData`. Add add your changes to that method as well.